src/secure-postgresql-db/index.ts

```diff
@@ -0,0 +1,551 @@
+    this._props.additionalSecurityGroups = additionalSecurityGroups;
+    return this;
+  }
+
+  /**
+  * Initial allocated storage for the DB.
+  * https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_rds.DatabaseInstance.html#allocatedstorage
+  * @param allocatedStorage
+  * @default 50
+  * @returns SecurePostgresqlDb
+  */
+  public allocatedStorage(allocatedStorage: number): SecurePostgresqlDb {
+    this._props.allocatedStorage = allocatedStorage;
+    return this;
+  }
+
+  /**
+  * Maximum allocated storage for the DB.
+  * https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_rds.DatabaseInstance.html#maxallocatedstorage
+  * @param maxAllocatedStorage
+  * @default 200
+  * @returns SecurePostgresqlDb
+  */
+  public maxAllocatedStorage(maxAllocatedStorage: number): SecurePostgresqlDb {
+    this._props.maxAllocatedStorage = maxAllocatedStorage;
+    return this;
+  }
+
+  /**
+  * Whether the database should be Multi-AZ.
+  * https://aws.amazon.com/rds/features/multi-az/
+  * @param multiAz
+  * @default false
+  * @returns SecurePostgresqlDb
+  */
+  public multiAz(multiAz: boolean): SecurePostgresqlDb {
+    this._props.multiAz = multiAz;
+    return this;
+  }
+
+  /**
+  * Backup plan rule configuration.
+  * https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_backup.BackupPlanRuleProps.html
+  * @param backupPlanRuleProps
+  * @default {
+      completionWindow: Duration.hours(4),
+      startWindow: Duration.hours(1),
+      // Midnight, every night
+      scheduleExpression: Schedule.cron({
+        day: '*',
+        hour: '0',
+        minute: '0',
+      }),
+      deleteAfter: Duration.days(7),
+    }
+  * @returns SecurePostgresqlDb
+  */
+  public backupPlanRuleProps(backupPlanRuleProps: BackupPlanRuleProps): SecurePostgresqlDb {
+    this._props.backupPlanRuleProps = backupPlanRuleProps;
+    return this;
+  }
+
+  /**
+  * Properties for configuring alarm thresholds.
+  * @param cpuUtilizationThresholdPct - Max CPU utilization percent. Alarm triggers if value goes above threshold.
+  * @param minFreeMemoryMB - Minimum free memory threshold, in MB. Alarm triggers if value goes below threshold.
+  * @param maxOpenConnections - Max open connections threshold. Alarm triggers if value goes above threshold.
+  * @param freeStorageThresholdPct - Free storage percent threshold. Alarm triggers if value goes below threshold.
+  * @default cpuUtilizationThresholdPct - 90
+  * @default minFreeMemoryMB - 1024
+  * @default maxOpenConnections - 100
+  * @default freeStorageThresholdPct - 20
+  * @returns SecurePostgresqlDb
+  */
+  public alarmProps(props: AlarmProps): SecurePostgresqlDb {
+    this._props.alarmProps = props;
+    return this;
+  }
+
+  /**
+  * The Database will be created using the provided snapshot.
+  * @param snapshotArn
+  * @returns SecurePostgresqlDb
+  */
+  public snapshotArn(snapshotArn: string): SecurePostgresqlDb {
```

Comment:
Can we change this method name to be "restoreFromSnapshot(snapshotArn: string)"? 
Also update the following with an example, 

```md 
 If you want to restore your database from an existing snapshot, specificy the snapshot value. 
 *Careful, some important notes:* 
 - Once you specify a snapshot to restore from, you must not remove it and continue to use that even if you are chaniging other properties associated with the database. 
  - Only during the initial setup, the data present in the snapshot is used, there after for all updates the data is not reset. 
    - If you remove the snapshotARN you have used previously, the current database will be reset to start from an empty snapshot and you will loose all existing data in the database. 
   - if you specify a different snapshotARN, your current database will be reset to start from that data present in the newly specified snapshot and you will loose all the existing data in your database. 
 
```

Reply:
Updated name and added notes